src/dune_rules/lib.ml

```diff
@@ -1421,6 +1435,9 @@ end = struct
           let open Memo.Build.O in
           match (dep : Lib_dep.t) with
           | Re_export (loc, name) ->
+            if is_bigarray name && not bigarray_in_std_libraries
```

Comment:
Some parts (like the end of the `get` function) are a bit difficult to follow because the bigarray check is intertwined with the rest of the logic. I'll refactor that part to see if we can extract the bigarray logic in a more precise way.